pkg/service/service.go

```diff
@@ -0,0 +1,133 @@
+package service
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	applabels "github.com/redhat-developer/odo/pkg/application/labels"
+	componentlabels "github.com/redhat-developer/odo/pkg/component/labels"
+	"github.com/redhat-developer/odo/pkg/occlient"
+)
+
+// ComponentInfo holds all important information about one component
+type ServiceInfo struct {
+	Name   string
+	Type   string
+	Status string
+}
+
+// ListCatalog lists all the available service types
+func ListCatalog(client *occlient.Client) ([]string, error) {
+
+	clusterServiceClasses, err := client.GetClusterServiceClassExternalNames()
+	if err != nil {
+		return []string{}, errors.Wrapf(err, "Unable to get cluster serviceClassExternalName")
+	}
+
+	return clusterServiceClasses, nil
+}
+
+// CreateService create new service from serviceCatalog
+func CreateService(client *occlient.Client, serviceName string, serviceType string, applicationName string) error {
+	labels := componentlabels.GetLabels(serviceName, applicationName, true)
+	// save service type as label
+	labels[componentlabels.ComponentTypeLabel] = serviceType
+	err := client.CreateServiceInstance(serviceName, serviceType, labels)
+	if err != nil {
+		return errors.Wrap(err, "unable to create service instance")
+
+	}
+	return nil
+}
+
+// DeleteService will delete the service with the provided `name`
+func DeleteService(client *occlient.Client, name string, applicationName string, projectName string) error {
+
+	labels := componentlabels.GetLabels(name, applicationName, false)
+	err := client.DeleteServiceInstance(labels)
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve list of services")
+	}
+	return nil
+
+}
+
+// List lists all the deployed services
+func List(client *occlient.Client, applicationName string, projectName string) ([]ServiceInfo, error) {
+	applicationSelector := fmt.Sprintf("%s=%s", applabels.ApplicationLabel, applicationName)
+	serviceNames, err := client.GetServiceStatus(projectName, componentlabels.ComponentLabel, applicationSelector)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to list services")
+	}
+
+	services := []ServiceInfo{}
+
+	for _, service := range serviceNames {
+		ctype, err := GetServiceType(client, service.Name, applicationName, projectName)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to list services")
+		}
+		services = append(services, ServiceInfo{Name: service.Name, Type: ctype, Status: service.Status})
+	}
+
+	return services, nil
+}
+
+// GetServiceType returns type of service in given application and project
+func GetServiceType(client *occlient.Client, serviceName string, applicationName string, projectName string) (string, error) {
+
+	// filter according to service and application name
+	selector := fmt.Sprintf("%s=%s,%s=%s", componentlabels.ComponentLabel, serviceName, applabels.ApplicationLabel, applicationName)
```

Comment:
you should use `ConvertLabelsToSelector` https://github.com/redhat-developer/odo/blob/50ef7cda4a41eae5213e174bb83e7f2b0f7d32ce/pkg/util/util.go#L13